cachito/workers/nexus.py

```diff
@@ -367,21 +367,22 @@ def upload_raw_component(repo_name, directory, components, to_nexus_hoster=True)
     :raise CachitoError: if the upload fails
     """
     params = {"repository": repo_name}
-    payload = {"raw.directory": directory}
+    additional_data = {"raw.directory": directory}
+    payload = {}
     for index, component in enumerate(components):
         n = index + 1
-        payload[f"raw.asset{n}.filename"] = component["filename"]
+        additional_data[f"raw.asset{n}.filename"] = component["filename"]
         with open(component["path"], "rb") as f:
             payload[f"raw.asset{n}"] = f.read()
 
     try:
-        upload_component(params, payload, to_nexus_hoster)
+        upload_component(params, payload, to_nexus_hoster, additional_data)
     except CachitoError:
         log.exception("Failed to upload %r to the raw Nexus repository", components)
         raise
 
 
-def upload_component(params, payload, to_nexus_hoster):
+def upload_component(params, payload, to_nexus_hoster, additional_data=None):
```

Comment:
Should we document the `additional_data` argument and why it is needed (i.e. because requests does not support sending non-files in the file payload)?

Reply:
Good catch. Done :)